### ISSUE TITLE
TestContext.WorkDirectory not initialized during build process

### ISSUE THREAD

**Issue by oznetmaster** [nunit/nunit#2061]
```
        [TestCaseSource("WorkDirectorySource")]
        public void TestCaseSourceCanAccessWorkDirectory(string workDirectory)
        {
            Assert.That(workDirectory, Is.EqualTo(_workDirectory));
        }

        static IEnumerable<string> WorkDirectorySource()
        {
            yield return TestContext.CurrentContext.WorkDirectory;
        }
```

This test appears to fail if the WorkDirectory is not the same as the CurrentDirectory.

The new code in TestContext:

```
        public string WorkDirectory
        {
            get { return _testExecutionContext?.WorkDirectory ??
#if PORTABLE
                    @"\My Documents"; }
#else
                    Directory.GetCurrentDirectory(); }
#endif
        }
```

returns the CurrentDirectory during the build phase since there is no TestExecutionContext (null) at the time that the TestCaseSource is processed/created.

**Comment by oznetmaster** [comment]
This was a breaking change that resulted from PR #2055.

**Comment by oznetmaster** [comment]
It probably need a test for this situation as well.

**Comment by oznetmaster** [comment]
I have been exploring how to fix this.  It seems like one easy possibility would be to create a dummy TestExecutionContext before the build phase so that all of the "static" values like WorkDirectory would be properly available.

**Comment by oznetmaster** [comment]
The same problem exists for TestContext.CurrentContext.TestDirectory

**Comment by CharliePoole** [comment]
The TestDirectory is fixed for each assembly.

**Comment by rprouse** [comment]
Fixed by #2072
